.github/workflows/model-evaluation.yml

```diff
@@ -0,0 +1,201 @@
+          AZURE_ADLS_GEN2_STORAGE_ACCOUNT: ${{ vars.AZURE_ADLS_GEN2_STORAGE_ACCOUNT }}
+          AZURE_ADLS_GEN2_FILESYSTEM_PATH: ${{ vars.AZURE_ADLS_GEN2_FILESYSTEM_PATH }}
+          AZURE_ADLS_GEN2_FILESYSTEM: ${{ vars.AZURE_ADLS_GEN2_FILESYSTEM }}
+
+        steps:
+          #--------------------------Setup--------------------------------
+
+          - name: Checkout
+            uses: actions/checkout@v4
+
+          - name: Setup python
+            uses: actions/setup-python@v5
+            with:
+              python-version: ${{ matrix.python_version }}
+              architecture: x64
+    
+          - name: Install azd
+            uses: Azure/setup-azd@v1.0.0
+    
+          - name: Install Nodejs
+            uses: actions/setup-node@v4
+            with:
+              node-version: 18
+
+          #--------------------------Deploy--------------------------------
+          - name: Check credentials
+            run: |
+              echo "AZURE_CLIENT_ID: ${{ env.AZURE_CLIENT_ID }}"
+              echo "AZURE_TENANT_ID: ${{ env.AZURE_TENANT_ID }}"
+              echo "AZURE_SUBSCRIPTION_ID: ${{ env.AZURE_SUBSCRIPTION_ID }}"
+              echo "AZURE_PRINCIPAL_ID: ${{ secrets.AZURE_PRINCIPAL_ID }}"
+
+          - name: Azure login
+            uses: azure/login@v2
+            with:
+              client-id: ${{ env.AZURE_CLIENT_ID }}
+              tenant-id: ${{ env.AZURE_TENANT_ID }}
+              subscription-id: ${{ env.AZURE_SUBSCRIPTION_ID }}
+            
+          - name: Azure CLI script
+            uses: azure/cli@v2
+            with:
+              azcliversion: latest
+              inlineScript: |
+                az account show
+
+          #--------------------------Run Evaluation--------------------------------
+          - name: Install dependencies
+            working-directory: ./evaluation
+            run: |
+              python -m pip install --upgrade pip
+              pip install -r scripts/requirements.txt
+              pip install pyrit
+          - name: Run AI Rag Evaluations
+            working-directory: ./evaluation
+            run: |
+              OPENAI_HOST=$OPENAI_HOST OPENAI_GPT_MODEL=$OPENAI_GPT_MODEL AZURE_OPENAI_EVAL_DEPLOYMENT=$AZURE_OPENAI_EVAL_DEPLOYMENT AZURE_OPENAI_SERVICE=$AZURE_OPENAI_SERVICE AZURE_SEARCH_SERVICE=$AZURE_SEARCH_SERVICE AZURE_SEARCH_INDEX=$AZURE_SEARCH_INDEX AZURE_PRINCIPAL_ID=$AZURE_PRINCIPAL_ID python -m scripts evaluate --config=./config.json --numquestions=10
+              echo "EVALUATION_RESULTS=evaluation/results/$(ls ./results/ | grep "experiment" | tail -n 1)" >> $GITHUB_ENV
+            env:
+              OPENAI_HOST: ${{ vars.OPENAI_HOST }}
+              OPENAI_GPT_MODEL: ${{ vars.OPENAI_GPT_MODEL }}
+              AZURE_OPENAI_EVAL_DEPLOYMENT: ${{ vars.AZURE_OPENAI_EVAL_DEPLOYMENT }}
+              AZURE_OPENAI_SERVICE: ${{ vars.AZURE_OPENAI_SERVICE }}
+              AZURE_SEARCH_SERVICE: ${{ vars.AZURE_SEARCH_SERVICE }}
```

Comment:
Many of those variables have been set at the job level, there's no point in repeating them here. Please check for duplicates and avoid repetitions!

Reply:
Got it, I'll clean up these duplicates.